src/velo-fed/VeloFarmerV3.sol

```diff
@@ -0,0 +1,715 @@
+     * @param dolaAmount Amount of DOLA to be added as liquidity in Velodrome DOLA/USDC pool
+     * @param usdcAmount Amount of USDC to be added as liquidity in Velodrome DOLA/USDC pool
+     */
+    function deposit(uint dolaAmount, uint usdcAmount) public onlyChair {
+        uint lpTokenPrice = getLpTokenPrice();
+
+        DOLA.approve(address(router), dolaAmount);
+        nUSDC.approve(address(router), usdcAmount);
+        (uint dolaSpent, uint usdcSpent, uint lpTokensReceived) = router.addLiquidity(address(DOLA), address(nUSDC), true, dolaAmount, usdcAmount, 0, 0, address(this), block.timestamp);
+        require(lpTokensReceived > 0, "No LP tokens received");
+
+        uint totalDolaValue = dolaSpent + (usdcSpent *DOLA_USDC_CONVERSION_MULTI);
+
+        uint expectedLpTokens = totalDolaValue *1e18 / lpTokenPrice *(PRECISION - maxSlippageBpsLiquidity) / PRECISION;
+        if (lpTokensReceived < expectedLpTokens) revert LiquiditySlippageTooHigh();
+        
+        uint lpBalance = LP_TOKEN_NATIVE.balanceOf(address(this));
+        LP_TOKEN_NATIVE.approve(address(dolaGauge), lpBalance);
+        dolaGauge.deposit(lpBalance);
+    }
+
+
+    /**
+     * @notice Calls `deposit()` with entire DOLA & USDC token balance of this contract.
+     */
+    function depositAll() external {
+        deposit(DOLA.balanceOf(address(this)), nUSDC.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Withdraws `dolaAmount` worth of LP tokens from gauge. Then, redeems LP tokens for DOLA/USDC.
+     * @dev If attempting to remove more DOLA than total LP tokens are worth, will remove all LP tokens.
+     * @param dolaAmount Desired dola value to remove from DOLA/USDC pool. Will attempt to remove 50/50 while allowing for `maxSlippageBpsLiquidity` bps of variance.
+     * @return Amount of USDC received from liquidity removal. Used by withdrawLiquidityAndSwap wrapper.
+     */
+
+    function withdrawLiquidity(uint dolaAmount) public onlyChair returns (uint) {
+        uint lpTokenPrice = getLpTokenPrice();
+        uint liquidityToWithdraw = dolaAmount *1e18 / lpTokenPrice;
+        uint ownedLiquidity = dolaGauge.balanceOf(address(this));
+
+        if (liquidityToWithdraw > ownedLiquidity) liquidityToWithdraw = ownedLiquidity;
+        dolaGauge.withdraw(liquidityToWithdraw);
+   
+        LP_TOKEN_NATIVE.approve(address(router), liquidityToWithdraw);
+        (uint amountUSDC, uint amountDola) = router.removeLiquidity(address(nUSDC), address(DOLA), true, liquidityToWithdraw, 0, 0, address(this), block.timestamp);
+
+        uint totalDolaReceived = amountDola + (amountUSDC *DOLA_USDC_CONVERSION_MULTI);
+
+        if ((dolaAmount *(PRECISION - maxSlippageBpsLiquidity) / PRECISION) > totalDolaReceived) {
+            revert LiquiditySlippageTooHigh();
+        }
+
+        return amountUSDC;
+    }
```

Comment:
Consider making internal and adding `withdrawLiquidity(uint dolaAmount) external onlyChair returns (uint);` function.